code/game/objects/items/bodybag.dm

```diff
@@ -89,4 +89,73 @@
 	// Make it possible to escape from bodybags in morgues and crematoriums
 	if(loc && (isturf(loc) || istype(loc, /obj/structure/morgue) || istype(loc, /obj/structure/crematorium)))
 		if(!open())
-			to_chat(user, "<span class='notice'>It won't budge!</span>")
\ No newline at end of file
+			to_chat(user, "<span class='notice'>It won't budge!</span>")
+
+/obj/item/bodybag/cryobag
+	name = "stasis bag"
+	desc = "A folded, non-reusable bag designed for the preservation of an occupant's body by stasis. Activates on occupant's unconscious state. Can be locked with medical ID."
+	icon = 'icons/obj/cryobag.dmi'
+	icon_state = "bodybag_folded"
+
+/obj/item/bodybag/cryobag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/cryobag/R = new /obj/structure/closet/body_bag/cryobag(user.loc)
+	R.add_fingerprint(user)
+	R.desc = desc
+	qdel(src)
+
+/obj/structure/closet/body_bag/cryobag
+	name = "stasis bag"
+	icon = 'icons/obj/cryobag.dmi'
+	item_path = /obj/item/bodybag/cryobag
+	var/used = 0
+	var/locked = 0
+	req_access = list(access_medical)
+
+/obj/structure/closet/body_bag/cryobag/dump_contents()
+	for(var/atom/movable/AM in src) //To prevent something standing on body_bag being affected
+		if(!ismob(AM))
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 0
+		used = 1	//We consider bag as used only if there is actual human in it
+	..()
+
+/obj/structure/closet/body_bag/cryobag/open()
+	. = ..()
+	if(used) //happens after dump_contents()
+		var/obj/item/O = new/obj/item(src.loc)

```

Comment:
Are you putting a ghost item inside the cryobag just so it can't be folded up once used?